types/block.go

```diff
@@ -741,6 +740,160 @@ func (cs *CommitSig) FromProto(csp tmproto.CommitSig) error {
 
 //-------------------------------------
 
+// ExtendedCommitSig is similar to CommitSig, except it also contains the vote
+// extension and vote extension signature related to this commit.
+type ExtendedCommitSig struct {
```

Comment:
This depends heavily on whether this structure will ever be serialized to JSON. I could go ahead and refactor it all now, but if we ever need to serialize these structures to JSON we'll have to refactor it all back.

Reply:
I've now refactored this, but have embedded the `CommitSig` within the `ExtendedCommitSig` directly for convenience. It seems somewhat more ergonomic to do it this way to me.

Reply:
Thanks @thanethomson, it now resembles the first version of the `ExtendedCommitSig` struct that I discussed informally with @williambanfield, where he convinced me to use two separate types here.
@williambanfield, could you please double-check that embedding `CommitSig` like this is OK for you?

One last question: since we are embedding without a field name, isn't `CommitSig`'s method `BlockID` also available as a method of `ExtendedCommitSig`? Is this wanted?

Reply:
It definitely works, I generally try to avoid embedding structs unless I have a good reason, like I'm using the methods of some struct to satisfy an interface. Otherwise, it adds a layer of indirection without a _ton_ of benefit. You do cut down on having to write the type name when calling its methods, but you have to dig around to understand which field's method is actually being called when you encounter a struct with an embedded field. On this type of tradeoff, I'll take explicit pretty much every time.

I think embedding sort of feels like inheritance, but frankly go is not really very good at inheritance. You _can_ use this language feature to do crude overloading, but because go doesn't provide the keywords and structures for actual inheritance to explicitly describe complex object relationships, (`abstract`, `overrides`, `implements` etc), it becomes difficult to trace what's actually being called so I try to avoid it.

Reply:
> What's the drawback of being able to access the inner CommitSig's fields as though they belonged directly to the ExtendedCommitSig?

I see no drawback in accessing the inner fields directly. Actually, I saw your latest changes (mostly deletes) and find it more elegant than the previous version.

My question here was rather on the "inherited" method `BlockID`

On the more general question with which I started this discussion, if William is not really against it, I also prefer how it looks like now.